src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -354,13 +354,24 @@ PAL_TlsIo AppleCryptoNative_SslRead(SSLContextRef sslContext, uint8_t* buf, uint
     return OSStatusToPAL_TlsIo(status);
 }
 
-int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore)
+int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore, int32_t* pOSStatus)
 {
+    if (pOSStatus == NULL)
+        return -1;
+
     if (sslContext == NULL || notBefore == NULL)
+    {
+        *pOSStatus = errSecInvalidData;
         return -1;
+    }
+
     if (cfHostname == NULL)
+    {
+        *pOSStatus = errSecInvalidData;
```

Comment:
Is it possible the OS will return errSecInvalidData as an actual error code?  I'm wondering what happens if it does: will it be confusing in that we won't know whether it came from the OS or whether it came from invalid arguments we pass into this function?